src/SessionManager.ts

```diff
@@ -161,7 +160,7 @@ export class SessionManager {
 
     const stateLink = session.links.find((l: any) => l.rel === 'state')
 
-    return new Promise(async (resolve, reject) => {
+    return new Promise(async (resolve) => {
```

Comment:
This `new Promise` thing is actually pointless - `waitForSession` is marked as `async`, so it will already return a `Promise`. We can just remove this line and replace all the `resolve`s with `return`s.

Reply:
agree